### DISCUSSION TITLE
renovate's dynamic configuration of project rules

### DISCUSSION THREAD

**Discussion by MIO1100** [renovatebot/renovate#22929] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I want to dynamically configure the rules for the repository. How can I use the resulting config if it is not in the main branch of the repository? I don't want to commit into this main branch. 

### What you tried so far.

I tried to use option "--use-base-branch-config". This option can merge configuration of main branch and other baseBranches. If you have not config at main branch, renovate can not use config from base branch.

**Comment by rarkins** [comment, marked as answer]
useBaseBranchConfig is the only way. Either that or you test locally using platform=local

**Reply by MIO1100** [reply]
Thanks for your answer, but when i run
`renovate --use-base-branch-config=merge`

and use this config 
`module.exports = {
  labels: ["dependencies"],
  endpoint: 'https://gitlab-test.com/api/v4/',
  platform: 'gitlab',
  repositories: ['my-repo'],
  baseBranches: ["test-branch"]
};`
renovate does not create issue and branches in the repository `my-repo`
renovate.json stay at this repo branch - `test-branch`

Maybe I misunderstood the description in the documentation about "--use-base-branch-config" function ?
May be I must have renovate.json file at master branch? I thing it is a wrong way

**Reply by rarkins** [reply]
You need a renovate config in the default branch

**Reply by MIO1100** [reply]
Thanks! I included a empty configuration file in the my default branch
`{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"]
}`
Thanks!!
